# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/FetchEventOpProxyParent.cpp
# Commit: cf972fe0d961
# Full Hash: cf972fe0d96164788f998f04008930474526e400
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-26 21:36:36
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Create IPC for transferring NavigationPreload's PerformanceTimingData. r=dom-worker-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138251
# ==============================================================================

diff -r f8afd25bf41e -r cf972fe0d961 dom/serviceworkers/FetchEventOpProxyParent.cpp
--- a/dom/serviceworkers/FetchEventOpProxyParent.cpp	Sat Feb 26 08:40:25 2022 +0000
+++ b/dom/serviceworkers/FetchEventOpProxyParent.cpp	Sat Feb 26 08:40:26 2022 +0000
@@ -121,9 +121,15 @@
   ParentToChildServiceWorkerFetchEventOpArgs copyArgs(aArgs.common(),
                                                       Nothing());
   if (aArgs.preloadResponse().isSome()) {
-    // Convert the preload response to ParentToChildInternalResponse.
-    copyArgs.preloadResponse() = Some(ToParentToChild(
-        aArgs.preloadResponse().ref(), WrapNotNull(aManager->Manager())));
+    // Convert the preload response to ParentToChildResponseWithTiming.
+    ParentToChildResponseWithTiming response;
+    response.response() =
+        ToParentToChild(aArgs.preloadResponse().ref().response(),
+                        WrapNotNull(aManager->Manager()));
+    response.timingData() = aArgs.preloadResponse().ref().timingData();
+    response.initiatorType() = aArgs.preloadResponse().ref().initiatorType();
+    response.entryName() = aArgs.preloadResponse().ref().entryName();
+    copyArgs.preloadResponse() = Some(response);
   }
 
   FetchEventOpProxyParent* actor =
@@ -135,11 +141,16 @@
   // need to add it to the arguments. Note that we have to make sure that the
   // arguments don't contain the preload response already, otherwise we'll end
   // up overwriting it with a Nothing.
-  Maybe<ParentToParentInternalResponse> preloadResponse =
+  Maybe<ParentToParentResponseWithTiming> preloadResponse =
       actor->mReal->OnStart(WrapNotNull(actor));
   if (copyArgs.preloadResponse().isNothing() && preloadResponse.isSome()) {
-    copyArgs.preloadResponse() = Some(ToParentToChild(
-        preloadResponse.ref(), WrapNotNull(aManager->Manager())));
+    ParentToChildResponseWithTiming response;
+    response.response() = ToParentToChild(preloadResponse.ref().response(),
+                                          WrapNotNull(aManager->Manager()));
+    response.timingData() = preloadResponse.ref().timingData();
+    response.initiatorType() = preloadResponse.ref().initiatorType();
+    response.entryName() = preloadResponse.ref().entryName();
+    copyArgs.preloadResponse() = Some(response);
   }
 
   IPCInternalRequest& copyRequest = copyArgs.common().internalRequest();